src/server/transaction.cc

```diff
@@ -498,8 +498,12 @@ bool Transaction::RunInShard(EngineShard* shard, bool txq_ooo) {
   // If we're the head of tx queue (txq_ooo is false), we remove ourselves upon first invocation
   // and successive hops are run by continuation_trans_ in engine shard.
   // Otherwise we can remove ourselves only when we're concluding (so no more hops will follow).
-  bool remove_txq = is_concluding || !txq_ooo;
+  // In case of multi transaction is_concluding represents only if the current running op is
```

Comment:
But also that doesn't break the if the current transaction is not a multi transaction and `is_concluding` is `true` ?

Reply:
should_release = not_multi && is_concluding